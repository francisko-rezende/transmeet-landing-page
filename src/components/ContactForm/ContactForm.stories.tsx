import React from 'react'

import { ComponentMeta, ComponentStory } from '@storybook/react'

import ContactForm from '.'

export default {
  title: 'ContactForm',
  component: ContactForm
} as ComponentMeta<typeof ContactForm>

//const Template: ComponentStory<typeof ContactForm> = (args) => <ContactForm {...args} />

const Template: ComponentStory<typeof ContactForm> = () => <ContactForm />

export const Default = Template.bind({})
