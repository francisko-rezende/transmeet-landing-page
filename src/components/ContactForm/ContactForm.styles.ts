import styled, { css } from 'styled-components'

export const Form = styled.form`
  margin-top: 41px;
  width: 100%;
  max-width: 430px;
  display: flex;
  flex-direction: column;
  gap: 20px;

  @media (max-width: 610px) {
    margin-top: 20px;
  }
`

export const Label = styled.label`
  z-index: 2;
  position: absolute;
  left: 11px;
  top: -5px;
  padding: 0 5px;
  width: fit-content;

  ${({ theme }) => css`
    background-color: ${theme.colors.typographyColors.white};
    color: ${theme.colors.typographyColors.gray};
    font-weight: ${theme.font.weight.semiBold};
    font-size: ${theme.font.sizes['2']};
  `}
`

export const TextAreaWrapper = styled.div`
  width: 100%;
  position: relative;
`
export const TextArea = styled.textarea`
  padding: 15px 23px;
  max-height: 173px;
  max-width: 470px;
  border-width: 1px;
  border-color: ${({ theme }) => theme.colors.typographyColors.gray};
  border-style: solid;
  border-radius: 4px;
  width: 100%;

  &:focus {
    outline: none;
    background-color: ${({ theme }) => theme.colors.neutralColors.grayLighter};
  }
`

export const SubmitButton = styled.button`
  font-size: 1.6rem;
  padding: 10px 0;
  cursor: pointer;
  margin-top: 80px;

  ${({ theme }) => css`
    background: ${theme.colors.secondaryColor.secondary};
    border: none;
    border-radius: 24px;
    color: ${theme.colors.typographyColors.white};
    font-weight: ${theme.font.weight.bold};

    &:hover {
      background: ${theme.colors.secondaryColor.secondaryDark};
    }

    @media (max-width: 610px) {
      margin-top: 20px;
    }
  `}
`
export const EmailInfoParagraph = styled.p`
  font-size: 1rem;
  text-align: center;
  margin-top: 40px;
  margin-bottom: 87px;
  ${({ theme }) => css`
    color: ${theme.colors.typographyColors.gray};
  `};

  @media (max-width: 610px) {
    margin-top: unset;
    margin-bottom: 30px;
  }
`
export const EmailLink = styled.a`
  text-decoration: underline;
  ${({ theme }) => css`
    color: ${theme.colors.typographyColors.gray};
    font-weight: ${theme.font.weight.bold};
  `}
`
