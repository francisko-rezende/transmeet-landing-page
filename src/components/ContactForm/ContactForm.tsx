import React from 'react'

import * as S from './ContactForm.styles'

const ContactForm = () => {
  return (
    <S.Wrapper>
      <h1>ContactForm</h1>
    </S.Wrapper>
  )
}

export default ContactForm
