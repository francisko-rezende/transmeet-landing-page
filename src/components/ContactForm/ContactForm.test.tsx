import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import ContactForm from '.'

describe('<ContactForm />', () => {
  test('should render the heading', () => {
    renderWithTheme(<ContactForm />)

    expect(
      screen.getByRole('button', { name: /enviar mensagem/i })
    ).toBeInTheDocument()
  })
})
