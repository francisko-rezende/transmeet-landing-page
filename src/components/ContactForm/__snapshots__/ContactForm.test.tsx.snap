// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    ContactForm
  </h1>
</main>
`
